libraries/AP_Winch/AP_Winch_Servo.cpp

```diff
@@ -0,0 +1,95 @@
+#include <AP_Winch/AP_Winch_Servo.h>
+
+extern const AP_HAL::HAL& hal;
+
+void AP_Winch_Servo::init(const AP_WheelEncoder* wheel_encoder)
+{
+    _wheel_encoder = wheel_encoder;
+}
+
+void AP_Winch_Servo::update()
+{
+    // return immediately if no servo is assigned to control the winch
+    if (!SRV_Channels::function_assigned(SRV_Channel::k_winch)) {
+        return;
+    }
+
+    if (!_servo_out_init) {
+        // set servo output range
+        SRV_Channels::set_angle(SRV_Channel::k_winch,  1000);
+        _servo_out_init = true;
+    }
+
+    // return immediately if no wheel encoder
+    if (_wheel_encoder == nullptr) {
+        return;
+    }
+
+    // if not doing any control stop output pwm exit
+    if (config.state == AP_Winch::STATE_DISABLED) {
+        SRV_Channels::set_safety_limit(SRV_Channel::k_winch, SRV_Channel::SRV_CHANNEL_LIMIT_ZERO_PWM);
```

Comment:
I might not be familiar enough with the SRV_Channels stuff, but isn't this a one time setup aspect that could be set in the `if (!_servo_out_init) {` block? It's also worth noting that this is running a ioctl every time you hit it, which is probably not what you want to be doing in flight.

Reply:
yes, that was a typo, fixed in a follow up commit, thanks!